src/hooks/tradingView/useChartLines.ts

```diff
@@ -189,59 +227,192 @@ export const useChartLines = ({
     });
   }, [stringGetter, currentMarketId, currentMarketPositionData, maybeDrawPositionLine]);
 
+  // Cache for order modification that stores the new orders that are submitted but not yet placed
+  const pendingOrderAdjustmentsRef = useRef<{
+    [clientId: string]: { orderPayload: HumanReadablePlaceOrderPayload; oldOrderId: string };
+  }>({});
+
+  const removePendingOrderAdjustment = (clientId: number) => {
+    const { [clientId]: removed, ...withoutOrderId } = pendingOrderAdjustmentsRef.current;
+    pendingOrderAdjustmentsRef.current = withoutOrderId;
+  };
+
+  const addPendingOrderAdjustment = (
+    orderPayload: HumanReadablePlaceOrderPayload,
+    oldOrderId: string
+  ) => {
+    pendingOrderAdjustmentsRef.current = {
+      ...pendingOrderAdjustmentsRef.current,
+      [orderPayload.clientId]: { orderPayload, oldOrderId },
+    };
+  };
+
+  const dispatch = useAppDispatch();
+
+  const onMoveOrderLine = useCallback(
+    async (order: SubaccountOrder, orderLine?: IOrderLineAdapter) => {
+      if (!orderLine) return;
+
+      const oldPrice = order.triggerPrice ?? order.price;
+      const newPrice = orderLine.getPrice();
+
+      // TODO(tinaszheng): do validation here for new price
```

Comment:
this will be in a followup PR bcuz i need to create some new redux state for handling local error notifications